### PR TITLE
Add LDAP page

### PR DESCRIPTION
**Pull request by evilnick**
Sorry @hyperbolic2346 it was easier to create a new PR

I'll mail you the diff if you like!

### CODE REVIEW

File: pages/k8s/ldap.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+sidebar: k8smain-sidebar
+permalink: ldap.html
+layout: [base, ubuntu-com]
+toc: False
+---
+
+There is a distinction between authentication and authorisation:
+
+  * Authentication verifies who a user is.
+  * Authorization deals with what a user is allowed to do.
```
Comment (tvansteenburgh) [inline pages/k8s/ldap.md]: Authorisation

File: pages/k8s/ldap.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+## Access the dashboard
+
+You can determine the web address for the OpenStack dashboard by running:
+
+```bash
+juju status openstack-dashboard
+```
+
+Open the address in a webbrowesr and login with the token obtained previously.
```
Comment (tvansteenburgh) [inline pages/k8s/ldap.md]: web browser

File: pages/k8s/troubleshooting.md [hunk 3]
```diff
@@ -209,3 +209,197 @@ This is caused by the API load balancer not forwarding ports in the context of t
...
+
+Testing the steps is important to determine the cause of the problem.
+
+### Can you communicate with Keystone and get an authorization token.
+
+First is to verify that Keystone communication works from both your client and
+the kubernetes-worker machines. The easiest thing to do here is to copy the
+kube-keystone.sh script to the machines of interest from kubernetes-master with
+`juju scp kubernetes-master/0:kube-keystone.sh .`, edit the script to include
+you credentials, `source kube-keystone.sh` and then run `get_keystone_token`.
```
Comment (tvansteenburgh) [inline pages/k8s/troubleshooting.md]: your

File: pages/k8s/troubleshooting.md [hunk 4]
```diff
@@ -209,3 +209,197 @@ This is caused by the API load balancer not forwarding ports in the context of t
...
+This will produce a token from the Keystone server. If that isn't working,
+check firewalls settings and your Keystone server. Note that the
```
Comment (tvansteenburgh) [inline pages/k8s/troubleshooting.md]: should this be "firewall settings on your Keystone server"?

File: pages/k8s/troubleshooting.md [hunk 5]
```diff
@@ -209,3 +209,197 @@ This is caused by the API load balancer not forwarding ports in the context of t
...
+kube-keystone.sh script could be overwritten, so it is a best practice to make
+a copy somewhere and use that.
+
+### Are the pods for Keystone authentication up and running properly
```
Comment (tvansteenburgh) [inline pages/k8s/troubleshooting.md]: Add ? at the end

File: pages/k8s/troubleshooting.md [hunk 6]
```diff
@@ -209,3 +209,197 @@ This is caused by the API load balancer not forwarding ports in the context of t
...
 By default there is no log aggregation of the Kubernetes nodes, each node logs locally. Please read over the [logging](../logging) page for more information.
+
+## Troubleshooting Keystone/LDAP issues
+
+The following section offers some notes to help determine issues with using Keystone
+for authentication/authorisation.
+
+Testing the steps is important to determine the cause of the problem.
+
+### Can you communicate with Keystone and get an authorization token.
```
Comment (tvansteenburgh) [inline pages/k8s/troubleshooting.md]: Add ? at the end

File: pages/k8s/troubleshooting.md [hunk 7]
```diff
@@ -209,3 +209,197 @@ This is caused by the API load balancer not forwarding ports in the context of t
...
+I1121 05:02:02.893499       1 keystone.go:544] Kubernetes API client created, server version v1.12
+I1121 05:02:02.998944       1 keystone.go:93] ConfigMaps synced and ready
+I1121 05:02:02.999045       1 keystone.go:101] Starting webhook server...
+I1121 05:02:02.999262       1 keystone.go:155] ConfigMap created or updated, will update the authorization policy.
+I1121 05:02:02.999459       1 keystone.go:171] Authorization policy updated.
+```
+
+### Is the configmap with the policy correct?
+
+Check the configmap contents. The pods logs above would complain if the yaml
```
Comment (tvansteenburgh) [inline pages/k8s/troubleshooting.md]: pod

**Comment by evilnick** [comment]
fixed. thanks!

File: pages/k8s/troubleshooting.md (hunk 4 above)
Comment (hyperbolic2346) [reply pages/k8s/troubleshooting.md]: It is both your client and Keystone server really, but I think what we ended up on will be fine.

**Comment by hyperbolic2346** [comment]
@evilnick I would like to see a diff if that's still possible. Interested to see what I should be doing differently to help you some.

**Pull request merged**